### PR TITLE
Match api update

### PR DESCRIPTION
**Pull request by twagger**
This PR is about match API :

- Update of match API to be more compliant with the game websocket
- Creation of a new endpoint : /users/{id}/matches
- Global simplification of the API as some endpoints and dto / exceptions are not needed anymore

### CODE REVIEW

**Review by fmonbeig — APPROVED** [review]
Great ! Thanks for this new Api

File: srcs/server/src/user/user.service.ts [hunk 1]
```diff
@@ -35,6 +35,10 @@ export class UserService {
     achievements: true,
   };
 
+  readonly includedMatchRelations: object = {
+    players: true,
+  };
+
   readonly userStatus = Object.freeze({
     AWAY: 0,
```
Comment (lefoun) [inline srcs/server/src/user/user.service.ts:43]: when do you set the userStatus ? 

And how do you know that a user is away ?

**Review by lefoun — APPROVED** [review]
Looks good to me except userStatus in the user service.

File: srcs/server/src/user/user.service.ts (hunk 1 above)
Comment (twagger) [reply srcs/server/src/user/user.service.ts:43]: Oh you are right, I this this is obsolete and should be removed. I will check tomorrow and repush on the same PR if needed.

**Review by fmonbeig — APPROVED** [review]
All Good

File: srcs/server/src/match/match.service.ts [hunk 2]
```diff
@@ -35,124 +29,74 @@ export class MatchService {
...
-
-    // Calculate win probability for player 1 and player 2
-    const wp1: number = this._computeWinProbability(p1.eloRating, p2.eloRating);
-    const wp2: number = this._computeWinProbability(p2.eloRating, p1.eloRating);
-
-    // Create a new match (try catch for user connection fail)
+    const iLeft = createMatchDto.players.findIndex((p) => p.side === 0);
+    const iRight = createMatchDto.players.findIndex((p) => p.side === 1);
+    // Check sides
+    if (iLeft === -1 || iRight === -1) throw new BadRequestException();
```
[the remark this answers is not in the document]
Comment (twagger) [reply srcs/server/src/match/match.service.ts:51]: :+1:

File: srcs/server/src/user/user.service.ts (hunk 1 above)
Comment (twagger) [reply srcs/server/src/user/user.service.ts]: Done

**Pull request merged**